internal/controller/pause.go

```diff
@@ -109,18 +108,42 @@ func (pm *PauseModule) runPipelineIfSafe(ctx context.Context, restConfig *rest.C
 	return true, nil
 }
 
-func (pm *PauseModule) updatePipelineLifecycle(ctx context.Context, restConfig *rest.Config, pipeline *kubernetes.GenericObject, status string) error {
+func (pm *PauseModule) updatePipelineLifecycle(ctx context.Context, restConfig *rest.Config, pipeline *kubernetes.GenericObject, phase string) error {
+
+	err := withDesiredPhase(pipeline, phase)
+	if err != nil {
+		return err
+	}
+
+	err = kubernetes.UpdateCR(ctx, restConfig, pipeline, "pipelines")
+	if err != nil {
+		return err
+	}
+	return nil
```

Comment:
by calling `UpdateCR()` here, we get the latest `pipeline` with the revised ResourceVersion from K8S, so any subsequent call to do an update within this reconcile call will use the latest ResourceVersion and won't result in a conflict error.